src/components/stats/Index.jsx

```diff
@@ -0,0 +1,160 @@
+import React, { useState, useEffect } from 'react';
+import { Card, CardDeck } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import likes from '../../assets/images/article-likes.png';
+import dislike from '../../assets/images/dislike.png';
+import followersIcon from '../../assets/images/followers.png';
+import followingIcon from '../../assets/images/following.png';
+import articleWrittenIcon from '../../assets/images/articles-written.png';
+import articleReadIcon from '../../assets/images/article-read.png';
+import {
+  getArticlesWrittenRequest, getArticlesReadRequest,
+  getFollowersRequest, getFollowingsRequest,
+  getAllArticlesByUser, getReactions,
+} from '../../redux/actions/stats';
+
+const Stats = (props) => {
+  const {
+    articlesWritten, articlesRead,
+    followers, followings,
+    getFollowers, getFollowings,
+    getArticlesWritten, getArticlesRead,
+    getUserArticles, getArticleReactions,
+    authorId,
+  } = props;
+
+  const [reactions, setReactions] = useState({
+    likes: 0,
+    dislikes: 0
+  });
+
+  const requestArticlesAndReaction = async () => {
+    const articles = await getUserArticles(authorId);
+    if (articles) {
+      const newReactions = articles
```

Comment:
@kleva-j why do we have this much mapping, filtering and reducing here?

Reply:
Thanks, I'll refactor it immediately.

Reply:
@kleva-j I'm asking why we're doing that. What are you trying to achieve?

Reply:
I am trying to map the returned articles to add up all the likes and dislikes of all the articles.

Reply:
@9jaswag I have refactored this code, please let me have your comments.

Reply:
@kleva-j how is the data on the backend being returned?

Reply:
It returns an array of articles.
Each item inside the array is an object that contains the `articleReactions` property set as an array and each item contained in this array is an object containing the like status of the article and the `userid` of the author that liked/disliked the article.

Reply:
so, on the frontend, we're looping every article a user has ever reacted to, then counting the likes and dislikes? What happens when the user has reacted to a million articles?

The backend should handle this logic and simply return the number that corresponds to each stat, instead of looping on the frontend.

Reply:
Fixed